src/main/java/com/dropbox/core/oauth/DbxCredential.java

```diff
@@ -163,13 +163,14 @@ public boolean aboutToExpire() {
      * but just add host.
      * @param requestConfig request config used to make http request.
      * @param host which host to call, only used in internal testing.
+     * @param scope space-delimited scope list. Must be subset of scope in current oauth2 grant.
      * @return DbxRefreshResult which contains app key and app secret.
      * @throws DbxOAuthException If refresh failed becasue of invalid refresh parameter or
      * refresh token is revoked.
      * @throws DbxException If refresh failed for general errors.
      */
-    public DbxRefreshResult refresh(DbxRequestConfig requestConfig, DbxHost host) throws
-        DbxException {
+    public DbxRefreshResult refresh(DbxRequestConfig requestConfig, DbxHost host, String scope)
```

Comment:
Not sure exactly how this is set up but maybe adding a nullable annotation to this parameter would be helpful so people know it can be null.  Generally in Java it is bad practice to pass null into things so maybe adding this annotation or adding support for "" as equal to null

Reply:
This is unfortunate since this SDK wants to minimize the size and function count so we use as least as 3rd party dependencies as possible.

I think currently this nullable annotation only exist in 3rd party libs not in nature Java so it's hard to make it happen.